### DISCUSSION TITLE
ECS Works Pools Should Support Specifying Volumes for Flow Runs

### DISCUSSION THREAD

**Discussion by rjruizes** [PrefectHQ/prefect-aws#383] [Q&A] [answered]
With ECS Task blocks, you can supply EFS Volume parameters in JSON for the task definition like so
```js
TASK_DEFINITION = {
    ...
    "containerDefinitions": [
        {
            "name": "prefect",
            ...
            "mountPoints": [EFS_MOUNT_POINT],
        },
    ],
    "volumes": [EFS_VOLUME],
}
```
But no such variables exist for templating an ECS job.

# Expectation / Proposal
Volumes and mountPoints should be a JSON variable in the template

**Comment by desertaxle** [comment, marked as answer]
Thanks for the issue @rjruizes! Custom fields like this are available in the ECS worker via templating, so we recommend using the ECS worker for more configurability. To achieve the same customization in the `ECSTask` block, you can use the `task_customizations` field. Let us know if you have further questions about either of those approaches!

**Reply by rjruizes** [reply]
Ah I understand the templating better now. I modified the base template in the UI (under Advanced) for my ECS Worker Pool to allow for supplying volumes, since it's not built into prefect-aws base template. Example:

```
      "volumes": "{{ volumes }}",
...
      "volumes": {
        "title": "Volumes",
        "description": "",
        "type": "object",
        "additionalProperties": {
          "type": "string"
        },
        "default": [
          {
            "name": "efs-nibrs",
            "efsVolumeConfiguration": {
              "fileSystemId": "fs-abcdefghjijklmno"
            }
          }
        ]
      },
```
The UI under Basic then updated the form to show the change.
<img width="555" alt="Screenshot 2024-02-21 at 11 45 16 AM" src="https://github.com/PrefectHQ/prefect-aws/assets/2654709/1e2f7986-a994-4e62-aad4-843e0bbc6a4f">
